bindings/java/wasmedge-jni/lib/common.c

```diff
@@ -0,0 +1,348 @@
+    if(exClass == NULL) {
+        exitWithError(JVM_ERROR, "Exception class not found.");
+    }
+    (*env)-> ThrowNew(env, exClass, message);
+
+    exitWithError(JVM_ERROR, "Exception thrown for no class def");
+}
+
+void throwNoSuchMethodError(JNIEnv *env, char* methodName, char* sig) {
+    jclass exClass;
+    char *className = "java/lang/NoSuchMethodError";
+
+    char message[1000];
+
+    strcat(message, methodName);
+    strcat(message, sig);
+
+    if(exClass == NULL) {
+        throwNoClassDefError(env, message);
+    }
+
+    (*env)->ThrowNew(env, exClass, methodName);
+    exitWithError(JVM_ERROR, "Exception thrown for no such method");
+}
+
+
+jclass findJavaClass(JNIEnv* env, char * className) {
+    jclass class = (*env)->FindClass(env, className);
+
+    bool hasException = checkAndHandleException(env, "find class error");
+    if(hasException) {
+        return NULL;
+    }
+
+    if(class == NULL) {
+        throwNoClassDefError(env, className);
+    }
+    return class;
+}
+
+jmethodID findJavaMethod(JNIEnv* env, jclass class, char* methodName, char* sig) {
+    jmethodID jmethodId = (*env)->GetMethodID(env, class, methodName, sig);
+    return jmethodId;
+
+}
+
+void getClassName(JNIEnv* env, jobject obj, char* buff) {
+    jclass cls = (*env)->GetObjectClass(env, obj);
+
+// First get the class object
+    jmethodID mid = (*env)->GetMethodID(env, cls, "getClass", "()Ljava/lang/Class;");
+    jobject clsObj = (*env)->CallObjectMethod(env, obj, mid);
+    checkAndHandleException(env, "get class name error");
+
+// Now get the class object's class descriptor
+    cls = (*env)->GetObjectClass(env, clsObj);
+
+// Find the getName() method on the class object
+    mid = (*env)->GetMethodID(env, cls, "getName", "()Ljava/lang/String;");
+
+// Call the getName() to get a jstring object back
+    jstring strObj = (jstring)(*env)->CallObjectMethod(env, clsObj, mid);
+    checkAndHandleException(env, "get name error");
+
+// Now get the c string from the java jstring object
+    const char* str = (*env)->GetStringUTFChars(env, strObj, NULL);
+
+// Print the class name
+    strcpy(buff, str);
+
+// Release the memory pinned char array
+    (*env)->ReleaseStringUTFChars(env, strObj, str);
+}
+
+
+long getPointer(JNIEnv* env, jobject obj) {
+    jclass cls = (*env)->GetObjectClass(env, obj);
+
+    if (cls == NULL) {
+        exitWithError(JVM_ERROR, "class not found!");
+    }
+
+    jfieldID fidPointer = (*env)->GetFieldID(env, cls, "pointer", "J");
+    if(fidPointer == NULL) {
+        exitWithError(JVM_ERROR, "pointer filed not found!");
+    }
+    jlong value = (*env)->GetLongField(env, obj, fidPointer);
+    return value;
+}
+
+void setPointer(JNIEnv* env, jobject obj, jlong val) {
+    jclass cls = (*env)->GetObjectClass(env, obj);
+    jfieldID fidPointer = (*env)->GetFieldID(env, cls, "pointer", "J");
+    (*env)->SetLongField(env, obj, fidPointer, val);
+}
+
+void handleWasmEdgeResult(JNIEnv* env, WasmEdge_Result * result) {
```

Comment:
The `gradle test` failed currently, I think that might relate to the error from C++ not get converted to Java exception properly, then JUnit cannot handle the problem.

Reply:
I separate them into two sets
### success
- ASTModuleContextTest 
- ConfigureContextTest 
- FunctionTypeContextTest
- GlobalInstanceContextTest 
- GlobalTypeContextTest 
- ImportTypeContextTest 
- LoaderContextTest 
- MemoryInstanceTest 
- MemoryTypeContextTest
- ExportTypeContextTest
- StatisticsContextTest
- TableInstanceTest
- TableTypeContextTest
- ValidatorContextTest
- ValueTest
- WasmEdgeTest
- CompilerContextTest
### failed
- ModuleInstanceContextTest 
- ExecutorContextTest 
- BaseTest 
- StoreContextTest 
- WasmEdgeVMTest

Reply:
Here it's
```
> Task :test FAILED
Assertion failed: (This == nullptr), function Executor, file executor.h, line 95.
```

Reply:
All the test cases passed both in [ci workflow](://github.com/WasmEdge/WasmEdge/actions/runs/3277515842/jobs/5394895796) and my local Ubuntu dev box. Can you provide your evn configuration and build steps for me to reproduce this issue?

Reply:
Yes, my environment is quite unstable right now, just ignore this for now :). I will try to figure out the root cause.